Bookish/Bookish.DataAccess/LibraryService.cs

```diff
@@ -81,7 +87,26 @@ public void AddBook(string isbn, string title, string authors, int copies)
             });
         }
 
-        public bool BookExists(string isbn)
+        public bool IsIsbnInvalid(string isbn)
+        {
+            var stripped = isbn.Replace("-", "");
+            return BookExists(isbn) || stripped.Length != 13;
+        }
+
+        public IEnumerable<BarcodeImage> GetBarcodes(string isbn, int copies)
+        {
+            var sql = "SELECT LibraryBooks.bookId FROM LibraryBooks WHERE LibraryBooks.isbn = @isbn ORDER BY LibraryBooks.bookId DESC";
+            var newCopyIds = databaseConnection.Query<int>(sql, new { isbn }).Take(copies).Reverse();
+            return newCopyIds.Select(copyId => new BarcodeImage(GenerateBarcodeImage(copyId), copyId));
+        }
+
+        private static Image GenerateBarcodeImage(int copyId)
+        {
+            var barcode = new Barcode();
+            return barcode.Encode(TYPE.CODE39, copyId.ToString(), Color.Black, Color.White, 350, 200);
```

Comment:
I don't think this belongs here as it's not to do with db access. I would create a new service, e.g. BarcodeService, which uses the LibraryService to get any information it needs from the db, then creates the barcodes 🙂